File: java/base/src/main/vsdk/toolkit/environment/geometry/geometricProcessing/polygonTriangulation/monotoneDecomposition/_VertexChain.java
package vsdk.toolkit.environment.geometry.geometricProcessing.polygonTriangulation.monotoneDecomposition;

import vsdk.toolkit.common.linealAlgebra.Vector2Dd;

final class _VertexChain {
    Vector2Dd point = new Vector2Dd();
    int[] adjacentVertexIndices = new int[4];
    int[] chainNodeIndicesByAdjacency = new int[4];
    int adjacencySlotCount;
}

File: typescript/base/src/main/vsdk/toolkit/environment/geometry/geometricProcessing/polygonTriangulation/monotoneDecomposition/_VertexChain.ts
import { Vector2Dd } from "../../../../../common/linealAlgebra/Vector2Dd.js";
export class _VertexChain {
    public point = new Vector2Dd();
    public adjacentVertexIndices = new Int32Array(4);
    public chainNodeIndicesByAdjacency = new Int32Array(4);
    public adjacencySlotCount = 0;
}
